test/functions/bradc/forgotReturn.good

```diff
@@ -1 +1 @@
-forgotReturn.chpl:5: error: illegal use of function that does not return a value: 'foo'
+forgotReturn.chpl:9: error: unresolved call '<<(2, void)'
```

Comment:
I heard that @ronawho already brought this up on chat, but find myself wondering whether we should/could treat both "returns void expression" and "contains no returns" functions as having "void" return type (from a user perspective), yet having the compiler flag the latter case in order to generate error messages more like the old one here (my motivation being that I think it might catch an error of misunderstanding or mis-typing earlier in the code).

Without having put too much thought into it, it seems to me like an advantage would be that error messages might appear earlier and be clearer to understand (e.g., on line 5 rather than line 9); and that a disadvantage would be that we'd essentially have two flavors of void functions distinguished only by the presence/absence of return statements (which seems easy for the compiler to keep straight, via a flag, say -- but may be weird to purist users, esp. if they think they want/need to be able to declare both flavors of void explicitly, which I wouldn't want to support...).